src/main/java/com/pinterest/secor/util/orc/VectorColumnFiller.java

```diff
@@ -166,13 +165,28 @@ public void convert(JsonElement value, ColumnVector vect, int row) {
     }
 
     static class MapColumnConverter implements JsonConverter {
-        private JsonConverter[] childrenConverters;
+        private JsonConverter[] childConverters;
 
         public MapColumnConverter(TypeDescription schema) {
-            List<TypeDescription> kids = schema.getChildren();
-            childrenConverters = new JsonConverter[kids.size()];
-            for (int c = 0; c < childrenConverters.length; ++c) {
-                childrenConverters[c] = createConverter(kids.get(c));
+            assertKeyType(schema);
+
+            List<TypeDescription> childTypes = schema.getChildren();
+            childConverters = new JsonConverter[childTypes.size()];
+            for (int c = 0; c < childConverters.length; ++c) {
+                childConverters[c] = createConverter(childTypes.get(c));
+            }
+        }
+
+        /**
+         * Rejects non-string keys. This is a limitation imposed by JSON specifications that only allows strings
+         * as keys.
+         */
+        private void assertKeyType(TypeDescription schema) {
+            TypeDescription keyType = schema.getChildren().get(0);
+            String keyTypeName = keyType.getCategory().getName();
+            if (!keyTypeName.equals("string")) {
```

Comment:
Should it be case insensitive comparison?

Reply:
That’s a good point. Not sure if ORC schema is case sensitive. But it would be a good idea to perform case insensitive comparison here just to be safe.